services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/DynamoDbEnhancedAsyncClient.java

```diff
@@ -425,7 +426,7 @@ default CompletableFuture<Void> transactWriteItems(TransactWriteItemsEnhancedReq
      * tables.
      * @return a {@link CompletableFuture} of {@link Void}.
      */
-    default CompletableFuture<Void> transactWriteItems(Consumer<TransactWriteItemsEnhancedRequest.Builder> requestConsumer) {
+    default CompletableFuture<TransactWriteItemsEnhancedResponse> transactWriteItems(Consumer<TransactWriteItemsEnhancedRequest.Builder> requestConsumer) {
```

Comment:
This is backwards incompatible, we'd need to introduce a set of new methods. Probably `transactWriteItemsWithResponse`? Prior art [deleteItemWithResponse](https://github.com/aws/aws-sdk-java-v2/blob/master/services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/DynamoDbTable.java#L272)